### PR TITLE
Adds PDW. To replace Bulldogs as roundstart weapons

### PR DESCRIPTION
**Pull request by LanCartwright**
I coded this drunk give me a few hours to balance this if you don't like it.

### CODE REVIEW

**Comment by LanCartwright** [comment]
image
thank you nienhaus

**Comment by WJohn** [comment]
Why are we adding more bullet types do we not have enough already???

**Comment by LanCartwright** [comment]
If I used the existing AP ammo, it would be a little overpowered, don't you think?

As well as that, this weapon isn't meant to be reloaded, for now. It's the go-to weapon to defend yourself in an emergancy.

File: code/modules/projectiles/ammunition/magazines.dm [hunk 1]
```diff
@@ -161,6 +161,14 @@
 	max_ammo = 8
 	multiple_sprites = 2
 
+/obj/item/ammo_box/magazine/pdwdap
+	name = "PDW magazine (5.8x21mm DAP)"
```
Comment (WJohn) [inline code/modules/projectiles/ammunition/magazines.dm]: This should be called 5.8x22mm, or 5.8x21mm, whichever this gun is meant to use.

**Comment by WJohn** [comment]
Also, not a big fan of the sprites. I can't tell the gun's shape or anything, though I think I'll need a reference image for the origin of this thing. Besides that, the gun appears to be too dark in general, I can hardly tell the fill apart from the outline.

**Comment by Dorsisdwarf** [comment]
:-1: 
I like the idea of a one-mag mini-machinegun, but fuck a whole new ammo caliber and tacti/k/ool sprite.

**Comment by LanCartwright** [comment]
If you don't like the new ammo caliber, I can revert it to use the name caliber as an already existing AP ammo, however it will deal about 27 damage which is quite a bit more lethal. @Dorsisdwarf

**Comment by LanCartwright** [comment]
image

**Comment by Zilenan91** [comment]
So we're nerfing Op roundstart weapons and not making the other ones cheaper to compensate for these things that they will immediately replace because they suck? 15 damage 15 in a clip? Really man these are worthless

:-1:

**Comment by LanCartwright** [comment]
No. We're promoting the purchase of weapons in nuke rounds.


Also it's 15+8 armour penetration. So 23 total. The armour penetrating factor is the important area since these are far more effective against an armoured opponant.

**Comment by Zilenan91** [comment]
>14 TCs for a C20
>2 TCs for a magazine
>need at least 3 clips or you run out too fast
>20 minimum per op
>100 TCs for all 5
>Ops have almost no TCs left
>not enough for a borg or any kind of utility whatsoever
>die because you don't have emags or any kind of flexibility

Also I don't think you understand how AP works. It doesn't make you magically do more damage, the weapon does 15 damage, and it ignores 8% of the armor of anything it hits. It will never do more than 15 damage.

**Comment by WJohn** [comment]
8 armor penetration does NOT equal 8 extra damage. 8 AP reduces bullet resistance of armor by 8%, if applicable, otherwise it does nothing.

These are at best 15 damage per bullet, and I don't think ANY round should ever be below 20. It's a BULLET.

**Comment by LanCartwright** [comment]
is that how it works? i thought it worked similar to the claymore. thanks for letting me know. I'll go ahead and change this

**Comment by Zilenan91** [comment]
So you're not going to address the fact that Ops need to spend 100 TCs to have weapons that are even somewhat reliable

**Comment by LanCartwright** [comment]
new gun is now anti-personnel. this will highly discourage people without the right armour to chase after an operative.

**Comment by Amemelius** [comment]
Why can't we just have C-20R instead of a remarkably shitty basic weapon? I mean, weaponry is crazy expensive anyway.

**Comment by TechnoAlchemisto** [comment]
I like the PDW, but it should not replace the primary weapon, the c-20 is perfectly fitting to replace the primary. Forcing ops to buy guns without any sort of subsidy is beyond bad, maybe you haven't played that much nuke ops, it's hard to balance, but this is a much to steep nerf.

I'd support it being a sidearm, however unnecessary it may be.

**Comment by LanCartwright** [comment]
Reason why C20r was removed was because the metagame was c20r and energy shield making any attempt to combat them using energy based weapons basically impossible, melee was impossible as they knockdown in 2 shots, preventing stunbatons, and nobody in sec has roundstart ballistics unless they unload the riot shotgun and load in lethals at roundstart.

In fact this has 120 more potential maximum damage (600 vs 480) against the bulldog.

**Comment by Zilenan91** [comment]
Sooo 45 damage on a weapon that can't be reloaded? You know that sure makes this better than the bulldog yepyep it's so efficient at killing people

**Comment by Nienhaus** [comment]
l5cjq42
WJohn approved.

**Comment by Supermichael777** [comment]
that looks like it would be imposible to hold properly. look at pictures of guns sober

**Comment by LanCartwright** [comment]
you can still buy the bulldog for cheaper than the c20r if you want it that badly

also here's an mspaint of how you hold the pp2000

image

**Comment by AlexanderUlanH** [comment]
Encouraging purchase of op weapons is what the sketchkin was for.  Then we decided that was a bad idea, and decided to give them proper primary weapons again.

Now apparently we've gone full circle?

**Comment by Zilenan91** [comment]
Like I said, the cost of the weapons is entirely too high to ever be something you want to purchase.

**Comment by LanCartwright** [comment]
image

That can be handled later. This is currently adding the existance of a PDW.

Discussion on coderbus about this is ongoing.

**Comment by Zilenan91** [comment]
This isn't just adding the existence of a PDW, this is removing the Ops easy decent way to kill people and replacing it with a comparative piece of shit. What you're doing is giving the Ops 100 less TCs because they're gonna throw this out immediately because it sucks.

**Comment by LanCartwright** [comment]
I did not give ops 100 less TCs.

**Comment by Zilenan91** [comment]
Yes you did, you're requiring them to spend a minimum of 90-100 TCs on weapons that aren't total shit.

**Comment by AlexanderUlanH** [comment]
There's no need for another gun, projectile type, and ammo casing sitting around in the code.  If this doesn't serve some important need that can only be satisfied with a brand new gun, :-1: :100: 

And I'll make a personal appeal that you please, please, please not to try too hard to come up with one after the fact to justify getting this gun in the game.  We saw that with autorifles and after all the effort to balance them they still add nothing to the game.

**Comment by Zilenan91** [comment]
I would have zero problems with these if they were in the uplink for like 1 TC or something or there was one or two in the shuttle round start, but as it stands replacing these with bulldogs is just a massive nerf to Ops

**Comment by LanCartwright** [comment]
Autorifles were for sec. This is for NukeOp.

it is agreed that the bulldogs for nukeops is not satisfactory and it was agreed upon it would be a good idea to replace it.

How about Bulldogs for NukeOps at less than 10 TC? 8 is something I thought of off the top of my head though if you want it cheaper, I will consider it.

**Comment by Zilenan91** [comment]
It's all nuke weapons that need to be cheaper in general, Ops shouldn't go bankrupt from just buying the things they need to function.

**Comment by AlexanderUlanH** [comment]
We're fine in the gun department.  If there's fine-tuned rebalancing to do, we have plenty of guns that see no ingame use to do it with.  If we want ops to be using telecrystals to have weapons, remove the bulldogs from the locker, add them to the uplink, and adjust telecrystal counts so that ops can buy bulldogs if they want them, or other things if they don't.  Though, personally, I'd say ops using bulldogs is fine.  They're definitely one of the most interesting weapon types by virtue of the variety of viable ammo types.

Every time I look in bullet code, I cry, because there are so many unused bullet, casing, and gun variants.  Save a coder today, and stop coding new guns that just spit out different numbers of marginally different bullets.

**Comment by Zilenan91** [comment]
Op TCs as they are right now are probably fine if we just flat-out reduce the gun prices to 5 each.

**Comment by LanCartwright** [comment]
I'm actually quite insulted at the bullet comment since this is a first for any hollow point bullet hence the need for a new bullet type.

I'll take a look balancing prices tomorrow and balance them.

The problem with Bulldogs is it does not promote variety and people just buy the ammo bag. It's used as a replacement as a gun rather than a crutch or a last resort. Few choose real guns.

**Comment by Zilenan91** [comment]
That's because bulldogs are better simply because they're free. If guns didn't cost frigging 14 TCs people might buy them more.

**Comment by LanCartwright** [comment]
I'd like to break that meta.

I'll make a full NukeOp rebalance after work tomorrow.

**Comment by Amemelius** [comment]
> I'll make a full NukeOp rebalance after work tomorrow. 

The very fact you thought this PR was a good idea and the fact you didn't even know what AP did makes me worry for the viability of any sort of rebalance you propose.

**Comment by AlexanderUlanH** [comment]
>Damage = 45
>AP = 15

Wow so unique, a bullet that takes three shots to crit an unarmored assistant, three shots to crit a security officer, and three shots to crit the Captain through his armor.  That's so different from the nuke ops' current choices, "two shots to crit anyone", and "four shots to crit, and one more if the target has armor."

Don't get offended if you can help it.  I really appreciate the other things you've done, and I like the idea of ops' using viruses.  But this gun doesn't add anything to the game, and clutters up a file that's already cluttered with the corpses of ideas exactly like this one.

I'll say it again: there's already a non-viable nuke op weapon that was given to ops with the intention of forcing them to buy weapons, the sketchin.  If you want that, use it.

**Comment by Zilenan91** [comment]
It's -15 AP, does less damage through armor.

**Comment by LanCartwright** [comment]
Doesn't matter Amelius. I thought armour penetration had consistency I.e same as claymore and ninja katana armour penetration.

As for the gun, it was made to promote use of primary weapons and the stats were set up to optimally defend oneself against a close range target (assistant charge). The reverse armour penetration is to tell players that this is not a replacement to a real NukeOp gun and should be used sparingly.

**Comment by Zilenan91** [comment]
So you replace their actually decent guns with this and still want them to use their overpriced primaries...?

**Comment by LanCartwright** [comment]
Well, my intention was to promote purchasing but if the price is also an issue, I will look into it.

**Comment by AlexanderUlanH** [comment]
Sorry, 3 shots to crit an assistant and 4 to crit an officer.  Let me list the reasons why this isn't necessary to promote weapon purchasing:

* We already have the stechkin.
* If you want to promote purchasing, use the stechkin.
* That's what it's there for.

Once upon a time, coders decided nuke ops should buy their own guns, and gave them stechkins, intentionally weak secondary weapons, to make them purchase primaries.  They decided afterwards it was a bad idea.  
Maybe they want to change their mind.  But doing that does not require making a new gun.

**Comment by LanCartwright** [comment]
The problem I see with a Stechkin currently is it's lacklustre and people will still use it to save TC. Too often do players forgo buying anything and instead take what they have.

I see the stetchkin as the truly last resort, and lance this weapon as a go-to backup weapon to retake that gun the assistant disarmed from you, or a go-to weapon when you run out of ammo or are being chased by assistants.


I really don't want to keep Bulldogs as free-default as I'd really like to see variety.

I will be redoing the balance prices and may redo damage to follow up.

There's a huge power gap between the pistol and other operator guns. And that's what the PDW aims to bridge.

**Comment by PKPenguin321** [comment]
go back to stun/kill energy guns for nuke ops when?

**Comment by Cebasiz** [comment]
lain has a vendetta against nuke and this consolidates it for me.

without lowering the price of the other weapons this is essentially removing 5 C-20R worth of TC MINIMUM right off the bat. what the hell is the winrate of nuke? it can't be much higher than 50% in rounds above 30 pop right?

what is the purpose of all these crazy changes? at least kor's were SORT of thought out. jesus.

**Comment by Bawhoppen** [comment]
:-1: bulldog shotties are better for ops, more basic and less utility
shouldn't get your weapons for free

also we already technically have PDWs in the game aswell

**Comment by Dorsisdwarf** [comment]
CR-20 is basically the p90 right?

Anyway, the point is:

As it stands, bulldogs are completely, ludicrously overpowered. (my fault)
They need a nerf.
This sidesteps the point and fucks everything up

therefore, :-1:

**Comment by LanCartwright** [comment]
The C20r and C90gl are completely different.

The C20r:
20 rounds.
2 round burst.
65 stamina damage per shot.
20 damage.

The C90gl:
30 rounds + grenade.
3 round burst, grenade.
No stamina damage addition.
35 damage per shot.

The Bulldog is imbalanced because someone decided to give them slugs for free at roundstart. It's completely unreasonable to give them drum based shotguns loaded with slugs at roundstart in the first place. But it only gets worse as they can get incredible amounts of ammo for a ridiculously now price.


I don't know what you'd want to do, but you can't keep the Bulldog at roundstart and you can't replace it with a C20r.

Let me reiterate; you absolutely cannot balance the bulldog any further than it's current state as it is a carbon copy of a combat shotgun except it uses a drum/magazine loading system.

This isn't sidestepping the problem; it's taking the problem directly. I do not see a way to balance the Bulldogs any further and there is nothing appropriate to replace them with.

**Comment by Zilenan91** [comment]
Just remove the roundstart guns, make both the weapons cost 5, and add a bulk ammo dufflebag for C20s and we'd be good.

**Comment by Dorsisdwarf** [comment]
Hey here's a neat suggestion

nerf bulldog slug damage since like I've been saying on every PR they're only 60 brute due to me messing up the math

**Comment by tkdrg** [comment]
A good suggestion I heard on OOC was removing the roundstart bulldog and giving ops a gun voucher for a free primary (sniper/c20r/bulldog). I liked that idea.

**Comment by Zilenan91** [comment]
I'd love that. What we also need are more options for Ops to do stuff that aren't just shoot with gun. The foam grenade pr is a good step for that. Something else I'd love to see is a backpack flamethrower to be used with the elite hardsuit.

**Comment by Bawhoppen** [comment]
that voucher idea is a nice one

**Comment by CPTANT** [comment]
The voucher system is inflexible and requires every weapon to be equally strong.

There is nothing wrong with just giving ops more TC's to buy the guns they want if the guns aren't retardedly priced liked they are now.

Make guns cheaper, it's pretty pointless to buy more than 1 per person anyway.

**Comment by Bawhoppen** [comment]
Voucher system isn't inflexible for that reason
most guns aren't comparable entirely, plus if the guns have the same TC cost, they would have the same value for a voucher too

**Comment by tkdrg** [comment]
Conflicts. Also I'm not sure nuke ops need this right now, but the sprites are nice so I'd be willing to merge this as an admin-only thing if you add a version with a standard firing pin (like the `/unrestricted` variants that most other syndi weapons have).

**Comment by KorPhaeron** [comment]
Negative AP makes block code freak out, this shouldn't be merged as is

**Comment by LanCartwright** [comment]
That's bullshit Kor we just talked about it.

**Comment by KorPhaeron** [comment]
>1 day ago

Posted that before our conversation

**Comment by LanCartwright** [comment]
My bad Kor, I was pretty agitated at the time since people were talking about how "minus ap dont work".

**Comment by Cebasiz** [comment]
don't merge this, nukeop guns are perfect as they are right now seriously

it's totally unnecessary now that bundles exist and the nuke have more TC

close posthaste pls.

**Comment by tkdrg** [comment]
i agree but i like the sprites though that's why i said merge for admin only/summon guns
in fact i might just manual merge this tomorrow don't hold your breath though

**Comment by Cebasiz** [comment]
oh that's okay then

**Comment by Cheridan** [comment]
stale, reopen upon request

**Pull request closed without merging**